gtsam/hybrid/HybridFactorGraph.cpp

```diff
@@ -25,7 +25,7 @@
 namespace gtsam {
 
 /* ************************************************************************* */
-std::set<DiscreteKey> HybridFactorGraph::discreteKeys() const {
+DiscreteKeys HybridFactorGraph::discreteKeys() const {
```

Comment:
I disagree with this. There is no possible use for this method, as I can’t think of a use case where we rely on the order of a set of concatenation keys.

Reply:
I disagree. There are multiple methods that take `DiscreteKeys` as an argument in `discrete`. Having a convenient way to get the discrete key vector makes sense.

Reply:
It would then have to be private, as we would not want to expose an "intermediate-only" method in the API. But I see you reverted this change, so I think we're good now.

Reply:
Why would it be intermediate?

Take CartesianProduct for example. It takes DiscreteKeys and is used for enumerating through all the values of a DiscreteTree.